File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/patterns/stops/either/FullStopSelectionState.java
package rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either;


import java.util.HashMap;
import java.util.Map;

import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.StopRenderState;

/**
 * <p>DTO for the full selection state for stops.</p>
 * <p>Created by epicstar on 10/9/16.</p>
 * @author Jeremy Jao
 * @author Michael Antonacci
 */

public class FullStopSelectionState implements EitherStopState<Map<Integer, StopRenderState>> {

    private final Map<Integer, StopRenderState> stopRenderStateMap;

    private FullStopSelectionState(Map<Integer, StopRenderState> stopRenderStateMap) {
        this.stopRenderStateMap = new HashMap<>(stopRenderStateMap);
    }

    public static FullStopSelectionState create(Map<Integer, StopRenderState> stopRenderStateMap) {
        return new FullStopSelectionState(stopRenderStateMap);
    }

    public Map<Integer, StopRenderState> value() {
        return stopRenderStateMap;
    }
}

File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/patterns/stops/either/FullStopSelectionState.kt
package rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either


import java.util.HashMap

import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.StopRenderState

/**
 *
 * DTO for the full selection state for stops.
 *
 * Created by epicstar on 10/9/16.
 * @author Jeremy Jao
 * @author Michael Antonacci
 */

class FullStopSelectionState private constructor(stopRenderStateMap: Map<Int, StopRenderState>) : EitherStopState<Map<Int, StopRenderState>> {

    private val stopRenderStateMap: Map<Int, StopRenderState>

    init {
        this.stopRenderStateMap = HashMap(stopRenderStateMap)
    }

    override fun value(): Map<Int, StopRenderState> {
        return stopRenderStateMap
    }

    companion object {

        @JvmStatic
        fun create(stopRenderStateMap: Map<Int, StopRenderState>): FullStopSelectionState {
            return FullStopSelectionState(stopRenderStateMap)
        }
    }
}
